internal/fields/sql_test.go

```diff
@@ -219,17 +219,23 @@ func TestField_ValidateSQLType(t *testing.T) {
 	integer := int64(0)
 	cases := []struct {
 		In    interface{}
+		Tags  []string
 		Error bool
 	}{
 		{In: "foo", Error: false},
 		{In: int16(0), Error: false},
 		{In: &integer, Error: false},
 		{In: [4]byte{}, Error: true},
 		{In: ifaceBinaryMarshal{}, Error: true},
+		{In: ifaceBinaryMarshal{}, Tags: []string{"binary"}, Error: false},
+		{In: &ifaceBinaryMarshal{}, Tags: []string{"binary"}, Error: false},
+		// Non-pointer proto does not work because the Marshal() method has a pointer receiver.
```

Comment:
It'd probably be nice to have the other SQL types in here too

Reply:
let's do that as a follow up.